packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -205,131 +172,108 @@ export const Product = ({
     );
   }
 
-  // If the customer has any subscription plan that matches a plan for the
-  // selected product, then they are already subscribed.
-  const customerIsSubscribed =
-    !customer.error &&
-    !plans.error &&
-    customerSubscriptions &&
-    customerSubscriptions.some(customerSubscription =>
-      productPlans.some(plan => plan.plan_id === customerSubscription.plan_id)
+  // Only check for upgrade or existing subscription if we have a customer.
+  if (customer.result) {
```

Comment:
This is a slight logic change - we can skip checking both for existing subscription or upgrade offer if we don't have an existing customer.